docs/source/java/cdata.rst

```diff
@@ -0,0 +1,313 @@
+                    compiler = {"cpp11"},
+                    linkpath = {"/arrow/cpp/build/debug/"},
+                    link = {"arrow"}
+            )
+    )
+    public class CDataInterfaceLibraryConfig implements InfoMapper {
+        @Override
+        public void map(InfoMap infoMap) {
+        }
+    }
+
+.. code-block:: shell
+
+    // Compile our Java code
+    javac -cp javacpp-1.5.7.jar CDataInterfaceLibraryConfig.java
+
+    // Generate CDataInterfaceLibrary
+    java -jar javacpp-1.5.7.jar CDataInterfaceLibraryConfig.java
+
+    // Generate libjniCDataInterfaceLibrary.dylib
+    java -jar javacpp-1.5.7.jar CDataInterfaceLibrary.java
+
+    // Validate libjniCDataInterfaceLibrary.dylib created
+    otool -L macosx-x86_64/libjniCDataInterfaceLibrary.dylib
+    macosx-x86_64/libjniCDataInterfaceLibrary.dylib:
+            libjniCDataInterfaceLibrary.dylib (compatibility version 0.0.0, current version 0.0.0)
+            @rpath/libarrow.800.dylib (compatibility version 800.0.0, current version 800.0.0)
+            /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1200.3.0)
+            /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
+
+**Java Test**
+
+Let's create a Java class to Test C Data Interface from Java to C++:
+
+.. code-block:: java
+
+    import org.apache.arrow.c.ArrowArray;
+    import org.apache.arrow.c.ArrowSchema;
+    import org.apache.arrow.c.Data;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.BigIntVector;
+
+    public class TestCDataInterfaceV5 {
+
+        public static void main(String[] args) {
+            CDataInterfaceLibrary.ArrowSchema arrowSchema = new CDataInterfaceLibrary.ArrowSchema();
+            CDataInterfaceLibrary.export_int32_type(arrowSchema);
+
+            ArrowSchema arrow_schema = ArrowSchema.wrap(arrowSchema.address());
+            System.out.println("Java C Data - Schema Pointer = = " + Long.toHexString(arrowSchema.address()));
+
+            CDataInterfaceLibrary.ArrowArray arrowArray = new CDataInterfaceLibrary.ArrowArray();
+            CDataInterfaceLibrary.export_int32_array(arrowArray);
+
+            ArrowArray arrow_array = ArrowArray.wrap(arrowArray.address());
+            System.out.println("Java C Data - Array Pointer = " + Long.toHexString(arrowArray.address()));
+            System.out.println("Java C Data - Array Data Pointer Buffer array->buffers[1] = " + Long.toHexString(arrowArray.buffers(1).address()));
+
+            BufferAllocator allocator = new RootAllocator();
+            BigIntVector bigIntVector = (BigIntVector) Data.importVector(allocator, arrow_array, arrow_schema, null);
+            System.out.println("Java C Data - BigIntVector: " + bigIntVector);
+
+            CDataInterfaceLibrary.release_int32_type(arrowSchema);
+            CDataInterfaceLibrary.release_int32_array(arrowArray);
```

Comment:
Why aren't we using the release callback of the C Data Interface objects?

Reply:
ArrowArray and ArrowSchema release these after Data.importVector finalize